docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,180 @@
+is no longer load balanced across all zones.
+
+We need to take measures to prevent this change from happening immediately
+upon upgrade.
+
+In particular with Kubernetes, it's important to keep in mind that after
+MeshService is enabled we no longer generate our own VIP.
+So any requests to MeshService hostnames go to the same
+outbound listener as with requests to Kubernetes DNS names, the ClusterIP.
+
+###### Status quo
+
+- On Kubernetes there are two DNS names/outbound IP addresses:
+  - Traffic to `name_namespace_svc_port`/ClusterIP is load-balanced cross-zone
+  - Traffic to the `<kuma.io/service>.mesh`/VIP is load-balanced cross-zone
+
+###### `MeshService`
+
+- We now have one outbound IP address:
+  - Traffic to the generated hostname of the `MeshService` goes through the Cluster IP
+  - This cluster IP is no longer cross-zone load balanced
+
+## Considered Options
+
+* Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
+  the `MeshService {name: demo-app, namespace: kuma-demo, spec.ports[port == 5000]}`.
+  * we can drop this behavior when we drop `kuma.io/service` support
+* For cross-zone:
+  * don't generate MeshService by default
+  * have a setting to disable using MeshServices
+  * change the default behavior of MeshService from single-zone load balancing to cross-zone.
+    Users would need to manually switch over to local zone load-balancing after having
+    potentially migrated cross-zone services to `MeshMultiZoneService`.
+  * instead of changing the behavior of MeshService, instead also generate
+    MeshMultiZoneService by default, which would reintroduce the cross-zone load balancing behavior.
+
+## Decision Outcome
+
+* Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
+  the `MeshService `{name: demo-app, namespace: kuma-demo, spec.ports[port == 5000]}`.
+* Change `MeshService` to do cross-zone load balancing by default and have users
+  switch over to local zone plus `MeshMultiZoneServices`
+
+## Pros and Cons of the Options
+
+### Don't generate MeshServices by default
+
+Users would need to enable MeshService generation.
+
+#### Pros
+
+* One of the easiest, most straightforward option as far as stability and UX goes.
+
+#### Cons
+
+* Will be harder to motivate users to switch at all
+* None of the advantages of MeshService & HostnameGenerator "for free"
+* MeshServices need to be deleted if generation is enabled then disabled
+
+### Generate but don't enable by default
+
+Users would need to enable usage of MeshService resources.
+
+#### Pros
+
+* One of the easiest, most straightforward option as far as stability and UX goes.
+
+#### Cons
+
+* Will be harder to motivate users to switch at all
+* None of the advantages of MeshService & HostnameGenerator "for free"
+* Odd that MeshService objects exist but aren't used
+
+### Make cross-zone the default behavior
+
+The cross-zone behavior would be kept for the old, `<kuma.io/service>.mesh` VIP
```

Comment:
I don't understand here, AFAIK the clusterIP clusters are not cross-zone so why to we need to change behaviour?

Reply:
Why wouldn't they be cross-zone? They have the same clusters

https://kuma.io/docs/2.8.x/production/cp-deployment/multi-zone/#cross-zone-communication-details